argilla/docs/reference/argilla-server/configuration.md

```diff
@@ -0,0 +1,123 @@
+
+From the Argilla version `1.19.0`, you must set up the search engine manually to work with datasets. You should set the
+environment variable `ARGILLA_SEARCH_ENGINE=opensearch` or `ARGILLA_SEARCH_ENGINE=elasticsearch` depending on the backend you're using
+The default value for this variable is set to `elasticsearch`. The minimal version for Elasticsearch is `8.5.0`, and for Opensearch is `2.4.0`.
+Please, review your backend and upgrade it if necessary.
+
+!!! warning
+    For vector search in OpenSearch, the filtering applied is using a `post_filter` step, since there is a bug that makes queries fail using filtering + knn from Argilla.
+    See https://github.com/opensearch-project/k-NN/issues/1286
+
+    This may result in unexpected results when combining filtering with vector search with this engine.
+
+## Launching
+
+### Using a proxy
+
+If you run Argilla behind a proxy by adding some extra prefix to expose the service, you should set the `ARGILLA_BASE_URL`
+environment variable to properly route requests to the server application.
+
+For example, if your proxy exposes Argilla in the URL `https://my-proxy/custom-path-for-argilla`, you should launch the
+Argilla server with `ARGILLA_BASE_URL=/custom-path-for-argilla`.
+
+NGINX and Traefik have been tested and are known to work with Argilla:
+
+- [NGINX example](https://github.com/argilla-io/argilla/tree/main/examples/deployments/docker/nginx)
+- [Traefik example](https://github.com/argilla-io/argilla/tree/main/examples/deployments/docker/traefik)
+
+### with `uvicorn`
+
+Since the Argilla Server is built on FastAPI, you can launch it using `uvicorn`:
+
+```bash
+uvicorn argilla_server:app --port 6900
+```
+
+!!! {note}
+    For more details about FastAPI and uvicorn, see [here](https://fastapi.tiangolo.com/deployment/manually/#run-a-server-manually-uvicorn).
+
+    You can also visit the uvicorn official documentation [here](https://www.uvicorn.org/#usage).
+
+
+## Environment variables
+
+You can set the following environment variables to further configure your server and client.
+
+### Server
+
+#### FastAPI
+
+- `ARGILLA_HOME_PATH`: The directory where Argilla will store all the files needed to run. If the path doesn't exist it will be automatically created (Default: `~/.argilla`).
+
+- `ARGILLA_BASE_URL`: If you want to launch the Argilla server in a specific base path other than /, you should set up this environment variable. This can be useful when running Argilla behind a proxy that adds a prefix path to route the service (Default: "/").
+
+- `ARGILLA_CORS_ORIGINS`: List of host patterns for CORS origin access.
+
+- `ARGILLA_DOCS_ENABLED`: If False, disables openapi docs endpoint at _/api/docs_.
+
+- `ARGILLA_ENABLE_TELEMETRY`: If False, disables telemetry for usage metrics.
+
```

Comment:
I don't think it is. I am not up to date with the current config and migration options so that is why I asked you to review and judge the current docs.

Reply:
I added an `Authentication` section and info about `ARGILLA_AUTH_SECRET_KEY`.